harness/determined/gpu.py

```diff
@@ -22,6 +22,19 @@ class GPU(NamedTuple):
     memoryUtil: float
 
 
+warned_fields = set()
```

Comment:
what is this warned fields for? I didn't see it was used anywhere.

Reply:
It's used on line 32 and 33. Basically if we encounter a problem reading a certain metric, we'll log a warning. But we don't want to log the same warning again every time we hit that issue. We do a similar thing in a few other places already.